src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ProfileDataNode.cs

```diff
@@ -125,5 +125,10 @@ public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFacto
         {
             return Array.Empty<DependencyListEntry>();
         }
+
+        public override int CompareToImpl(ISortableNode other, CompilerComparer comparer)
+        {
+            return comparer.Compare(_methodNode, ((ProfileDataNode)other)._methodNode);
```

Comment:
Comparison is missing _targetDetails. I think this will probably not sort correctly if we use profile data when compiling (IBC).
You should probably add a test for this scenario too.

Reply:
@MichalStrehovsky

Reply:
Looking at the factory for ProfileDataNode, the method is the only key used to create the node. I don't think one can ever have two profile data nodes per method.

Reply:
Ah, good point. Thanks for confirming @MichalStrehovsky